lib/iris/coord_systems.py

```diff
@@ -292,37 +327,109 @@ def as_cartopy_projection(self):
         )
 
     def as_cartopy_globe(self):
-        return self._crs.globe
+        return self._globe
+
+    @cached_property
+    def _globe(self):
+        if self._datum is not None:
+            short_datum = _short_datum_names.get(self._datum, self._datum)
+            # Cartopy doesn't actually enact datums unless they're provided without
+            # ellipsoid axes, so only provide the datum
+            return ccrs.Globe(short_datum, ellipse=None)
+        else:
+            return ccrs.Globe(
+                ellipse=None,
+                semimajor_axis=self._semi_major_axis,
+                semiminor_axis=self._semi_minor_axis,
+            )
 
-    def __getattr__(self, name):
-        if name == "semi_major_axis":
+    @cached_property
```

Comment:
ditto here. I think what you are trying to do might be considered akin to a directed acylic graph (DAG) (like excel). you want `_crs` to be invalidated when `_globe` changes, but that doesn't happen with a `cached_property` afaik?

Reply:
Ah, I've just seen your _wipe_cached_properties. Does that handle the resetting in all cases?

Reply:
Maybe need some tests that change the datum / globe / crs?

Reply:
> Does that handle the resetting in all cases?

Yep

Reply:
> Maybe need some tests that change the datum / globe / crs?

Will do

Reply:
Done